sdk/appconfiguration/app-configuration/src/policies/throttlingRetryPolicy.ts

```diff
@@ -24,6 +25,23 @@ export function throttlingRetryPolicy(): RequestPolicyFactory {
   };
 }
 
+/**
+ * Maximum wait duration for the expected event to happen = `10000 ms`(default value is 10 seconds)(= maxWaitTimeInMilliseconds)
+ * Keep checking whether the predicate is true after every `1000 ms`(default value is 1 second) (= delayBetweenRetriesInMilliseconds)
+ */
+export async function checkWithTimeout(
```

Comment:
Maybe I should move this to core-utils?

Reply:
The entire throttling policy really should end up in the core-http (or similar) library.

The only real difference between this one and the 'official' is that our AppConfig policy reacts based on a thrown exception. The other one assumes that it'll just be part of the response with no thrown error - perhaps that never actually worked.

https://github.com/Azure/azure-sdk-for-js/blob/c4f51bee047ee069fb1a005d9132425372cec0c2/sdk/core/core-http/src/policies/throttlingRetryPolicy.ts#L48

Reply:
Might not be worth it if app-config is the only service that has retry-after-ms header with the error.

If we see at least one other service that does this, we can move this policy to the core?

Reply:
I'm wondering both ways - do we think the existing policy is ever actually used? 

I don't think it's our code that's doing the throw so I'm wondering if the core-http policy works at all. Perhaps this is a question we can answer by going through some swaggers at some point.